File: 二叉树的深度.java
public class Solution {
    public int TreeDepth(TreeNode root) {
            return TreeD(root,0);    
            }    
    private int TreeD(TreeNode root,int d){
           if (root==null)            
            return d;        
           else           
            d++;        
           return Math.max(TreeD(root.left,d),TreeD(root.right,d));
           }
}

File: 剑指offer（c++）/二叉树的深度.cpp
#include <iostream>
#include <stdlib.h>
#include <string>
using namespace std;
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(NULL), right(NULL) {
	}
};

class Solution {
    public:    
    int TreeDepth(TreeNode* pRoot){
        if(!pRoot) return 0 ;
        return max(1+TreeDepth(pRoot->left), 1+TreeDepth(pRoot->right));
   }
};

int main()
{
	TreeNode* root;
	root =(TreeNode*)malloc(sizeof(TreeNode));
	root -> val =0;
	root -> left=NULL;
	root -> right=NULL;
	Solution s;
	cout<<s.TreeDepth(root);
}
